source/elements/oneDAL/source/index.rst

```diff
@@ -1,128 +1,81 @@
+This document specifies requirements for implementations of Intel oneAPI Data Analytics Library (oneDAL).
 
-- Data Management provides interfaces for data acquisition, initial
-  preprocessing, data and model representation.
-- Algorithms provide interfaces for data analysis and data modeling,
-  and support batch, online and distributed computation modes.
-- Services include interfaces used across the library by Data
-  Management and Algorithms components: memory allocation, error
-  handling and interaction with underlying hardware. The programmer
-  can provide a queue specify the target device for algorithm
-  execution.
+oneDAL is a library
+that helps speed up big data analysis by providing highly optimized
+algorithmic building blocks for all stages of data analytics
+(preprocessing, transformation, analysis, modeling, validation, and
+decision making) in batch, online, and distributed processing modes of
+computation. The current version of oneDAL provides
+Data Parallel C++ (DPC++) API extensions to the traditional C++ interface.
 
-.. Much of oneDAL is defined only for the host processor.  The features are
-.. enabled to run on oneAPI devices.
+For general information, visit |github|.
 
-.. 1. **Data Management**: DPC++ numeric table
-.. 2. **Algorithms**: K-means, Covariance, Principal Component Analysis,
-..    Stochastic Gradient Descent, and Logistic Regression
-.. 3. **Services**: DPC++ execution context
+.. toctree::
+   :maxdepth: 3
+   :caption: Introduction
 
+   introduction/classical-ml.rst
+   introduction/classes.rst
+   introduction/building-blocks.rst
 
-.. code:: cpp
+.. toctree::
+   :maxdepth: 2
+   :caption: Programming model
 
-  #include "daal_sycl.h"
+   programming/model.rst
+   programming/algoritmics.rst
 
-  // Common oneDAL namespaces
-  using namespace daal::services;
-  using namespace daal::algorithms;
-  using namespace daal::data_management;
+.. toctree::
+   :maxdepth: 2
+   :caption: Data management
 
-  int main(int argc, char *argv[]) {
-    // Create a DPC++ queue with the desired device selector
-    cl::sycl::queue queue{cl::sycl::gpu_selector()};
+   data/numeric_tables.rst
+   data/structures.rst
 
-    // Create an execution context from the DPC++ queue and set up as the
-    // default for all algorithms
-    Environment::getInstance()->
-      setDefaultExecutionContext(SyclExecutionContext{queue});
+.. toctree::
+   :maxdepth: 3
+   :caption: Algorithms
 
-    // Create a DPC++ buffer from the data allocated on host
-    const size_t nRows = 10;
-    const size_t nCols = 5;
-    const float dataHost[] = {
-      0.42, -0.88,  0.46,  0.04, -0.86,
-      -0.74, -0.59,  0.42, -1.44, -0.40,
-      ...
-    };
-    cl::sycl::buffer<float, 1> dataBuffer{dataHost, nCols * nRows);
+   algorithms/modes.rst
+   algorithms/algorithm_inputs.rst
+   algorithms/algorithm_outputs.rst
+   algorithms/algorithm_descriptor.rst
+   algorithms/algorithm_processing.rst
```

Comment:
doesn't it make sense to create a section that these sections will be a part of? as it is done with computation modes, for example

Reply:
Once we will be adding content there we likely will do this